skyportal/handlers/api/analysis.py

```diff
@@ -909,9 +935,19 @@ def get(self, analysis_resource_type, analysis_id=None):
                 must be "obj" (more to be added in the future)
             - in: path
               name: analysis_id
-              required: true
+              required: false
               schema:
-                type: integer
+                type: int
+              description: |
+                ID of the analysis to return.
+            - in: query
+              name: objID
+              nullable: true
+              schema:
+                type: string
+              description: |
+                Return any analysis on an object with ID that has a (partial) match
+                to this argument (i.e., the given argument is "in" the object's ID).
```

Comment:
Are you sure there is partial matching? Because it looks like you are using exact matching on line 1007.